### PR TITLE
fix: use Either to handle both Activity and Fragment with the same interface

### PR DESCRIPTION
**Pull request by hotchemi**

### CODE REVIEW

File: ktx/src/main/java/permissions/dispatcher/ktx/PermissionsRequesterImpl.kt [hunk 1]
```diff
@@ -1,39 +1,81 @@
...
+        is FragmentType -> right.requireContext()
+    }
+
+private val ViewType.viewModelStoreOwner: ViewModelStoreOwner
+    get() = when (this) {
+        is ActivityType -> left
+        is FragmentType -> right
+    }
+
+private val ViewType.lifeCycleOwner: LifecycleOwner
```
Comment (mannodermaus) [inline ktx/src/main/java/permissions/dispatcher/ktx/PermissionsRequesterImpl.kt:71]:
```suggestion
private val ViewType.lifecycleOwner: LifecycleOwner
```

For consistency with the AndroidX property ☀️

File: sample/build.gradle [hunk 2]
```diff
@@ -18,6 +18,9 @@ android {
     sourceSets {
         main.java.srcDirs += "src/main/kotlin"
     }
+    lintOptions {
+        abortOnError false
```
Comment (hotchemi) [inline sample/build.gradle:22]: somehow the error occurs only on GitHub Action, not on my local 🤔 

https://github.com/permissions-dispatcher/PermissionsDispatcher/runs/2008862691?check_suite_focus=true#step:6:569

File: ktx/src/main/java/permissions/dispatcher/ktx/PermissionsRequesterImpl.kt (hunk 1 above)
Comment (hotchemi) [reply ktx/src/main/java/permissions/dispatcher/ktx/PermissionsRequesterImpl.kt:71]: thx bro! I'll think something over entirely and polish PR with description😄

**Comment by hotchemi** [comment]
just noticed another solution. let me close the PR

**Pull request closed without merging**